From 94633c36f9eb34e721faf38270b3dddc8f1cdaed Mon Sep 17 00:00:00 2001
From: Tom Rini <trini@konsulko.com>
Date: Tue, 2 Aug 2022 07:33:47 -0400
Subject: [PATCH] net: Make DM_ETH be selected by NETDEVICE

The deadline for DM_ETH migration passed 2 years ago.  Now that
platforms which cannot be migrated have been either removed or had
drivers disabled, and platforms that needed minor help to migrate have
been forcefully migrated, we can complete the migration.

This entails select'ing DM_ETH under NETDEVICES, and then removing now
extraneous depends on lines.  In a few places, we can now either remove
options or just simplify later dependencies.

Cc: Ramon Fried <rfried.dev@gmail.com>
Cc: Simon Glass <sjg@chromium.org>
Signed-off-by: Tom Rini <trini@konsulko.com>
Acked-by: Ramon Fried <rfried.dev@gmail.com>
---

 Makefile            |  1 +
 drivers/net/Kconfig | 78 +++++++++++++++++++++++++++++++++------------
 2 files changed, 58 insertions(+), 21 deletions(-)

diff --git a/Makefile b/Makefile
index f049d77dca..37215c499b 100644
--- a/Makefile
+++ b/Makefile
@@ -1147,6 +1147,7 @@ ifneq ($(CONFIG_SPL_FIT_GENERATOR),)
 endif
 	$(call deprecated,CONFIG_WDT,DM watchdog,v2019.10,\
 		$(CONFIG_WATCHDOG)$(CONFIG_HW_WATCHDOG))
+	$(call deprecated,CONFIG_DM_ETH,Ethernet drivers,v2020.07,$(CONFIG_NET))
 	$(call deprecated,CONFIG_DM_I2C,I2C drivers,v2022.04,$(CONFIG_SYS_I2C_LEGACY))
 	@# CFG_SYS_TIMER_RATE has brackets in it for some boards which
 	@# confuses this rule. Use if() to send just a single character which
diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index ba814a98ff..e89bdd00c1 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -6,7 +6,7 @@ config ETH
 	def_bool y
 
 config DM_ETH
-	bool
+	bool "Enable Driver Model for Ethernet drivers"
 	depends on DM
 	help
 	  Enable driver model for Ethernet.
@@ -17,7 +17,7 @@ config DM_ETH
 
 config DM_MDIO
 	bool "Enable Driver Model for MDIO devices"
-	depends on PHYLIB
+	depends on DM_ETH && PHYLIB
 	help
 	  Enable driver model for MDIO devices
 
@@ -42,7 +42,7 @@ config DM_MDIO_MUX
 
 config DM_DSA
 	bool "Enable Driver Model for DSA switches"
-	depends on DM_MDIO
+	depends on DM_ETH && DM_MDIO
 	depends on PHY_FIXED
 	help
 	  Enable driver model for DSA switches
@@ -93,7 +93,7 @@ config DSA_SANDBOX
 menuconfig NETDEVICES
 	bool "Network device support"
 	depends on NET
-	select DM_ETH
+	default y if DM_ETH
 	help
 	  You must select Y to enable any network device support
 	  Generally if you have any networking support this is a given
@@ -111,7 +111,7 @@ config PHY_GIGE
 
 config AG7XXX
 	bool "Atheros AG7xxx Ethernet MAC support"
-	depends on ARCH_ATH79
+	depends on DM_ETH && ARCH_ATH79
 	select PHYLIB
 	help
 	  This driver supports the Atheros AG7xxx Ethernet MAC. This MAC is
@@ -120,6 +120,7 @@ config AG7XXX
 
 config ALTERA_TSE
 	bool "Altera Triple-Speed Ethernet MAC support"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This driver supports the Altera Triple-Speed (TSE) Ethernet MAC.
@@ -152,7 +153,7 @@ config BCM_SF2_ETH_GMAC
 
 config BCM6348_ETH
 	bool "BCM6348 EMAC support"
-	depends on ARCH_BMIPS
+	depends on DM_ETH && ARCH_BMIPS
 	select DMA
 	select DMA_CHANNELS
 	select MII
@@ -162,7 +163,7 @@ config BCM6348_ETH
 
 config BCM6368_ETH
 	bool "BCM6368 EMAC support"
-	depends on ARCH_BMIPS
+	depends on DM_ETH && ARCH_BMIPS
 	select DMA
 	select MII
 	help
@@ -170,6 +171,7 @@ config BCM6368_ETH
 
 config BCMGENET
 	bool "BCMGENET V5 support"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This driver supports the BCMGENET Ethernet MAC.
@@ -178,13 +180,14 @@ source "drivers/net/bnxt/Kconfig"
 
 config CORTINA_NI_ENET
 	bool "Cortina-Access Ethernet driver"
-	depends on CORTINA_PLATFORM
+	depends on DM_ETH && CORTINA_PLATFORM
 	help
 	  This driver supports the Cortina-Access Ethernet MAC for
 	  all supported CAxxxx SoCs.
 
 config CALXEDA_XGMAC
 	bool "Calxeda XGMAC support"
+	depends on DM_ETH
 	help
 	  This driver supports the XGMAC in Calxeda Highbank and Midway
 	  machines.
@@ -208,6 +211,7 @@ config DM9000_USE_16BIT
 
 config DWC_ETH_QOS
 	bool "Synopsys DWC Ethernet QOS device support"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This driver supports the Synopsys Designware Ethernet QOS (Quality
@@ -304,7 +308,7 @@ config EEPRO100
 	  ethernet family of adapters.
 
 config ETH_SANDBOX
-	depends on SANDBOX
+	depends on DM_ETH && SANDBOX
 	default y
 	bool "Sandbox: Mocked Ethernet driver"
 	help
@@ -314,7 +318,7 @@ config ETH_SANDBOX
 	  This driver is particularly useful in the test/dm/eth.c tests
 
 config ETH_SANDBOX_RAW
-	depends on SANDBOX
+	depends on DM_ETH && SANDBOX
 	default y
 	bool "Sandbox: Bridge to Linux Raw Sockets"
 	help
@@ -334,6 +338,7 @@ config ETH_DESIGNWARE
 
 config ETH_DESIGNWARE_MESON8B
 	bool "Amlogic Meson8b and later glue driver for Synopsys Designware Ethernet MAC"
+	depends on DM_ETH
 	select ETH_DESIGNWARE
 	help
 	  This provides glue layer to use Synopsys Designware Ethernet MAC
@@ -344,7 +349,7 @@ config ETH_DESIGNWARE_SOCFPGA
 	select SYSCON
 	select DW_ALTDESCRIPTOR
 	bool "Altera SoCFPGA extras for Synopsys Designware Ethernet MAC"
-	depends on ETH_DESIGNWARE
+	depends on DM_ETH && ETH_DESIGNWARE
 	help
 	  The Altera SoCFPGA requires additional configuration of the
 	  Altera system manager to correctly interface with the PHY.
@@ -352,7 +357,7 @@ config ETH_DESIGNWARE_SOCFPGA
 
 config ETH_DESIGNWARE_S700
 	bool "Actins S700 glue driver for Synopsys Designware Ethernet MAC"
-	depends on ETH_DESIGNWARE
+	depends on DM_ETH && ETH_DESIGNWARE
 	help
 	  This provides glue layer to use Synopsys Designware Ethernet MAC
 	  present on Actions S700 SoC.
@@ -430,6 +435,7 @@ config FTMAC100
 
 config FTGMAC100
 	bool "Ftgmac100 Ethernet Support"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This driver supports the Faraday's FTGMAC100 Gigabit SoC
@@ -457,6 +463,7 @@ config SYS_DISCOVER_PHY
 
 config MCFFEC
 	bool "ColdFire Ethernet Support"
+	depends on DM_ETH
 	select PHYLIB
 	select SYS_DISCOVER_PHY
 	help
@@ -469,6 +476,7 @@ config SYS_UNIFY_CACHE
 
 config FSLDMAFEC
         bool "ColdFire DMA Ethernet Support"
+	depends on DM_ETH
 	select PHYLIB
 	select SYS_DISCOVER_PHY
 	help
@@ -480,6 +488,15 @@ config KS8851_MLL
 	help
 	  The Microchip KS8851 parallel bus external ethernet interface chip.
 
+if KS8851_MLL
+if !DM_ETH
+config KS8851_MLL_BASEADDR
+	hex "Microchip KS8851-MLL Base Address"
+	help
+	  Define this to hold the physical address of the device (I/O space)
+endif #DM_ETH
+endif #KS8851_MLL
+
 config KSZ9477
 	bool "Microchip KSZ9477 I2C controller driver"
 	depends on DM_DSA && DM_I2C
@@ -502,7 +519,7 @@ config MV88E6XXX
 config MVGBE
 	bool "Marvell Orion5x/Kirkwood network interface support"
 	depends on ARCH_KIRKWOOD || ARCH_ORION5X
-	select PHYLIB
+	select PHYLIB if DM_ETH
 	help
 	  This driver supports the network interface units in the
 	  Marvell Orion5x and Kirkwood SoCs
@@ -607,6 +624,7 @@ config OCTEONTX2_CGX_INTF
 
 config PCH_GBE
 	bool "Intel Platform Controller Hub EG20T GMAC driver"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This MAC is present in Intel Platform Controller Hub EG20T. It
@@ -668,14 +686,25 @@ config SJA1105
 config SMC911X
 	bool "SMSC LAN911x and LAN921x controller driver"
 
+if SMC911X
+
+if !DM_ETH
+config SMC911X_BASE
+	hex "SMC911X Base Address"
+	help
+	  Define this to hold the physical address
+	  of the device (I/O space)
+endif #DM_ETH
+
 config SMC911X_32_BIT
 	bool "Enable SMC911X 32-bit interface"
-	depends on SMC911X
 	help
 	  Define this if data bus is 32 bits. If your processor use a
 	  narrower 16 bit bus or cannot convert one 32 bit word to two 16 bit
 	  words, leave this to "n".
 
+endif #SMC911X
+
 config SUN7I_GMAC
 	bool "Enable Allwinner GMAC Ethernet support"
 	help
@@ -691,12 +720,14 @@ config SUN7I_GMAC_FORCE_TXERR
 
 config SUN4I_EMAC
 	bool "Allwinner Sun4i Ethernet MAC support"
+	depends on DM_ETH
 	select PHYLIB
 	help
 	  This driver supports the Allwinner based SUN4I Ethernet MAC.
 
 config SUN8I_EMAC
         bool "Allwinner Sun8i Ethernet MAC support"
+        depends on DM_ETH
         select PHYLIB
 	select PHY_GIGE
         help
@@ -719,6 +750,7 @@ config TULIP
 	  This driver supports DEC DC2114x Fast ethernet chips.
 
 config XILINX_AXIEMAC
+	depends on DM_ETH
 	select PHYLIB
 	select MII
 	bool "Xilinx AXI Ethernet"
@@ -726,7 +758,7 @@ config XILINX_AXIEMAC
 	  This MAC is present in Xilinx Microblaze, Zynq and ZynqMP SoCs.
 
 config XILINX_AXIMRMAC
-	depends on ARCH_VERSAL
+	depends on DM_ETH && ARCH_VERSAL
 	bool "Xilinx AXI MRMAC"
 	help
 	  MRMAC is a high performance, low latency, adaptable Ethernet
@@ -738,6 +770,7 @@ config VSC7385_ENET
 	bool "Vitesse 7385 Switch Firmware Upload driver"
 
 config XILINX_EMACLITE
+	depends on DM_ETH
 	select PHYLIB
 	select MII
 	bool "Xilinx Ethernetlite"
@@ -745,6 +778,7 @@ config XILINX_EMACLITE
 	  This MAC is present in Xilinx Microblaze, Zynq and ZynqMP SoCs.
 
 config ZYNQ_GEM
+	depends on DM_ETH
 	select PHYLIB
 	bool "Xilinx Ethernet GEM"
 	help
@@ -752,7 +786,7 @@ config ZYNQ_GEM
 
 config PIC32_ETH
 	bool "Microchip PIC32 Ethernet Support"
-	depends on MACH_PIC32
+	depends on DM_ETH && MACH_PIC32
 	select PHYLIB
 	help
 	  This driver implements 10/100 Mbps Ethernet and MAC layer for
@@ -760,7 +794,7 @@ config PIC32_ETH
 
 config GMAC_ROCKCHIP
 	bool "Rockchip Synopsys Designware Ethernet MAC"
-	depends on ETH_DESIGNWARE
+	depends on DM_ETH && ETH_DESIGNWARE
 	help
 	  This driver provides Rockchip SoCs network support based on the
 	  Synopsys Designware driver.
@@ -775,7 +809,7 @@ config RENESAS_ETHER_SWITCH
 
 config RENESAS_RAVB
 	bool "Renesas Ethernet AVB MAC"
-	depends on RCAR_64
+	depends on DM_ETH && RCAR_64
 	select PHYLIB
 	select PHY_ETHERNET_ID
 	help
@@ -793,7 +827,7 @@ config MPC8XX_FEC
 
 config SNI_AVE
 	bool "Socionext AVE Ethernet support"
-	depends on ARCH_UNIPHIER
+	depends on DM_ETH && ARCH_UNIPHIER
 	select PHYLIB
 	select SYSCON
 	select REGMAP
@@ -803,7 +837,7 @@ config SNI_AVE
 
 config SNI_NETSEC
 	bool "Socionext NETSEC Ethernet support"
-	depends on SYNQUACER_SPI
+	depends on DM_ETH && SYNQUACER_SPI
 	select PHYLIB
 	help
 	  This driver implements support for the Socionext SynQuacer NETSEC
@@ -892,6 +926,7 @@ config TSEC_ENET
 
 config MEDIATEK_ETH
 	bool "MediaTek Ethernet GMAC Driver"
+	depends on DM_ETH
 	select PHYLIB
 	select DM_GPIO
 	select DM_RESET
@@ -918,6 +953,7 @@ config HIFEMAC_MDIO
 
 config HIGMACV300_ETH
 	bool "HiSilicon Gigabit Ethernet Controller"
+	depends on DM_ETH
 	select DM_RESET
 	select PHYLIB
 	help
@@ -926,7 +962,7 @@ config HIGMACV300_ETH
 
 config FSL_ENETC
 	bool "NXP ENETC Ethernet controller"
-	depends on DM_MDIO
+	depends on DM_ETH && DM_MDIO
 	help
 	  This driver supports the NXP ENETC Ethernet controller found on some
 	  of the NXP SoCs.
-- 
2.45.2

